includes/promoted-jobs/class-wp-job-manager-promoted-jobs-status-handler.php

```diff
@@ -73,6 +68,33 @@ public function fetch_updates() {
 		}
 	}
 
+	/**
+	 * Checks if the update logic should be executed or not.
+	 *
+	 * @param int $current_time The current time.
+	 * @return bool True if the update logic should be executed, false otherwise.
+	 */
+	private function should_update( $current_time ) {
+		$interval = $this->get_current_interval();
+		if ( ! $interval ) {
+			// If the interval is not set or is zero, we don't update.
+			return false;
+		}
+		$last_execution = get_option( self::LAST_EXECUTION_OPTION_KEY, 0 );
```

Comment:
Not a big deal, but I think it's worth also casting it to `int`, similar to the `get_current_interval`.